src/engine/component.rb

```diff
@@ -28,10 +28,10 @@ def update(delta_time) end
     def destroy!
       destroy
       game_object.components.delete(self)
-      self.class.instance_variable_get(:@methods).each do |method|
-        singleton_class.send(:undef_method, method)
-        singleton_class.send(:define_method, method) { |*args, **kwargs| raise "This component has been destroyed" }
-      end
+      # self.class.instance_variable_get(:@methods).each do |method|
+      #   singleton_class.send(:undef_method, method)
+      #   singleton_class.send(:define_method, method) { |*args, **kwargs| raise "This component has been destroyed" }
+      # end
```

Comment:
This code is quite important as it stops you accessing stuff that's been destroyed. It forces you to check if something has been destroyed before using it as this could result in you accidentally using things that don't exist. An example would be if you didn't check for a light being destroyed and then still accessed the light's position, colour and size. You would see a light that shouldn't exist. I've fixed this on master along with various issues this casued such as the shield breaking the game when hitting and accessing asteroids that it shouldn't.

Reply:
Rebased and dropped that change, and things look to be working now 👍